micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetrics.java

```diff
@@ -295,6 +311,13 @@ enum GcGenerationAge {
                 put("partial gc", YOUNG);
                 put("global garbage collect", OLD);
                 put("Epsilon", OLD);
+                // GPGC
+                put("GPGC New", YOUNG); // old naming
+                put("GPGC Old", OLD); // old naming
+                put("GPGC New Cycles", YOUNG); // new naming
```

Comment:
Just for the sake of documentation:
This line (`GPGC New Cycles`) puzzled me a little. I built this branch locally and tested it. The meters + this map correspond what I saw in `MemoryPoolMXBean` except `GPGC New Cycles` is not listed in `getMemoryManagerNames`, see if I print out each `ManagementFactory.getMemoryPoolMXBeans()`:

```
GenPauseless New Gen (Heap memory) [GPGC New Pauses]
GenPauseless Old Gen (Heap memory) [GPGC Old Pauses]
GenPauseless Perm Gen (Heap memory) [GPGC Old Cycles]
CodeCache (Non-heap memory) [CodeCache]
Native Heap (Non-heap memory) [NativeHeap]
```

On the other hand, if I register a `NotificationListener`, I get these GC names (from GarbageCollectionNotificationInfo):
- `GPGC New Cycles`
- `GPGC New Pauses`
- `GPGC Old Cycles`
- `GPGC Old Pauses`

According to the [release notes](https://docs.azul.com/prime/release-notes#prime_stream_22_12_0_0) it is valid but I still find it weird as a GC name:
>New JMX MXBean metrics replace old metric name below `java.lang.GarbageCollector` to increase accuracy for GC monitoring added with [JDK-8265136](https://bugs.openjdk.org/browse/JDK-8265136): Previously, metric "GPGC New/Old" was providing a sum of GC pauses and concurrent GC duration. This metric is replaced by the following:
>- GPGC New/Old Cycles: duration time in ms of the concurrent GC which runs in parallel to application threads and is not stopping the application.
>- GPGC New/Old Pauses: GC pause time in ms.

Reply:
Indeed the motivation for these renaming was similar to what is described in JDK-8265136.
This inconsistence that you're pointing to is historical ;) But in this particular case, I believe, 'GPGC *' is what is used.